File: backend/python/pdfToImage.py
import sys
import os
from pdf2image import convert_from_path

# Get the PDF file and output folder from the command-line arguments
pdf_file = sys.argv[1]
output_folder = sys.argv[2]
image_format = sys.argv[3].lower()  # The format, e.g., 'jpg' or 'png'

# Ensure the output directory exists
os.makedirs(output_folder, exist_ok=True)

# Set path for poppler binaries (Windows only)
os.environ['PATH'] += r';C:\poppler-24.07.0\Library\bin'  # Adjust this to your poppler path

try:
    # Convert the PDF to images
    images = convert_from_path(pdf_file)

    # Map image formats for Pillow compatibility
    if image_format == 'jpg':
        image_format = 'JPEG'
    elif image_format == 'png':
        image_format = 'PNG'

    # Save each page as an image file
    for i, image in enumerate(images):
        image_file = os.path.join(output_folder, f"page_{i + 1}.{image_format.lower()}")
        image.save(image_file, image_format)
        print(f"Saved image: {image_file}")
    
    # print("success")
except Exception as e:
    print(f"error: {str(e)}")

File: backend/php/pdfToImage.php
<?php
if ($_SERVER['REQUEST_METHOD'] == 'POST') {
    $pdfFile = $_FILES['file']['tmp_name'];
    $imageFormat = "jpg"; // You can change this to 'png' if needed

    // Define the output folder
    $outputFolder = '../../files/images/'; // Ensure this directory exists or is writable

    // Ensure the output folder exists
    if (!file_exists($outputFolder)) {
        mkdir($outputFolder, 0777, true);
    }

    // Set up the Python script command
    $command = escapeshellcmd("python ../python/pdfToImage.py " . escapeshellarg($pdfFile) . " " . escapeshellarg($outputFolder) . " " . escapeshellarg($imageFormat));
    $output = shell_exec($command);

    // Capture the image file paths printed by the Python script
    $lines[] =  $output;
    $image_urls = [];


     // Iterate through the output lines and capture all image paths
     foreach ($lines as $line) {
        if (strpos($line, 'Saved image:') !== false) {
            // Extract the file path from the printed line
            $image_urls = trim(str_replace('Saved image:', '', $line));
        }
    }

    // // Check if any images were found
    if (isset($image_urls)) {
        echo json_encode(['success' => true, 'image_urls' => $image_urls]);
    } else {
        echo json_encode(['error' => 'No images were found after conversion']);
    }
}
